client/src/contexts/userContext.js

```diff
@@ -29,20 +60,80 @@ const UserProvider = (props) => {
           const data = await res.data;
 
           dispatch({ type: UPDATE_USER, payload: data });
+          history.push('/messenger');
         } else {
-          // if error then redirect to login
-          console.log('ERROR USER NOT FOUND');
-
-          dispatch({ type: USER_ERROR });
           history.push('/');
         }
       } catch (err) {
-        console.log('ERROR USER NOT FOUND');
-
-        dispatch({ type: USER_ERROR });
         history.push('/');
       }
     },
+    login: async (email, password) => {
+      try {
+        const res = await axios.post('/user/login', { email, password });
+        if (res && (res.status === 200 || res.status === 201)) {
+          await actions.fetchUser();
+          history.push('/messenger');
+        }
+
+        return res;
+      } catch (err) {
+        if (err.message.includes('400'))
+          dispatch({
+            type: USER_ERROR,
+            payload: { errorMsg: 'Invalid Credentials' },
+          });
+        if (err.message.includes('404'))
+          dispatch({
+            type: USER_ERROR,
+            payload: { errorMsg: 'User Not Found' },
+          });
+        if (err.message.includes('500'))
+          dispatch({
+            type: USER_ERROR,
+            payload: { errorMsg: 'Server Error' },
+          });
```

Comment:
could set a variable for errorMesg in each if statement and then have just one dispatch at the end

Reply:
I like that

Reply:
```
 let errorMsg;
        if (err.message.includes('400')) errorMsg = 'Invalid Credentials'   
        if (err.message.includes('404')) errorMsg = 'User not found'
        if (err.message.includes('500')) errorMsg = 'Server error'

          dispatch({
            type: USER_ERROR,
            payload: { errorMsg },
          });
```